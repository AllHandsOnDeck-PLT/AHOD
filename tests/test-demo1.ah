int i
series<Card> deck
when do series<Card> CREATEDECK(series<Card> deck):
{ 
    series<string> types 
    string type 
    Card card
    types = ["R0", "R1", "R2", "R3", "R4"]
    for (i = 0; i < types.size(); i = i + 1):
    {
        type = types[i]
        card = Card(type, true)   
        deck.push(card) 
    }
    return deck
} 
main:
{
    Card card
    /*card = Card("R0", true)*/
    /*deck = [card]*/
    deck = [Card("D0", true)]
    deck = do CREATEDECK(deck)
    for (i = 0 ; i < 5; i = i + 1):
    {
        card = deck[i]
        do PRINT(card.type)
        do PRINT(card.faceup)
    }
}
