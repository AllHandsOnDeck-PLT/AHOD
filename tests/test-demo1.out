D0
1
R0
1
R1
1
R2
1
R3
1
